code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -141,7 +160,11 @@ private void processEvents() {
             else if (EventType.GENERIC_TRACK.getName().equalsIgnoreCase(eventToProcess.getType()) &&
                     EventSource.REQUEST_CONTENT.getName().equalsIgnoreCase(eventToProcess.getSource())) {
                 // handle the track event information from the generic track request content event
-                track(eventToProcess);
+                track(eventToProcess, eventData);
```

Comment:
Can we extract event data in function track instead of passing as an argument. It will make code cleaner.

Reply:
`track` is also called from `processRulesEngineResponseEvent` where the eventData is passed in.